File: problems/2529_maximum_count_of_positive_integer_and_negative_integer/Solution.java
class Solution {
    public int countVowelStrings(int n) {
        int[] dp = {1,1,1,1,1};
        for (int i = 1; i < n; i++)
            for (int j = 1; j < 5; j++) dp[j] += dp[j-1];
        int sum = 0;
        for (int d : dp) sum += d;
        return sum;
    }
}

File: problems/2529_maximum_count_of_positive_integer_and_negative_integer/solution.py
class Solution:
    def countVowelStrings(self, n: int) -> int:
        vowels = 'aeiou'
        count = 0
        def dfs(idx, length):
            nonlocal count
            if length == n:
                count += 1
                return
            for i in range(idx, 5):
                dfs(i, length + 1)
        dfs(0, 0)
        return count
